vep-mapping-pipeline/repeat_expansion_variants/pipeline.py

```diff
@@ -24,33 +25,55 @@ def none_to_nan(*args):
-            # summary format which was used previously.
-            gene_symbols = clinvar_record.measure.preferred_gene_symbols
-            if not gene_symbols:
-                gene_symbols = ['-']
-            # Extract HGNC ID
-            hgnc_ids = clinvar_record.measure.hgnc_ids
-            hgnc_id = hgnc_ids[0] if len(hgnc_ids) == 1 and len(gene_symbols) == 1 else '-'
-            # Append data strings
-            for gene_symbol in gene_symbols:
-                variant_data.append([
-                    clinvar_record.measure.name,
-                    clinvar_record.accession,
-                    gene_symbol,
-                    hgnc_id
-                ])
+        if i and i % 100000 == 0:
+            total_repeat_expansion_variants = clinvar_xml_utils.ClinVarRecordMeasure.MS_REPEAT_EXPANSION + \
+                                              clinvar_xml_utils.ClinVarRecordMeasure.MS_NO_COMPLETE_COORDS
+            logger.info(f'Processed {i} records, collected {total_repeat_expansion_variants} repeat expansion variants')
+
+        # Skip a record if it does not contain variant information
+        if not clinvar_record.measure:
+            continue
+
+        # Repeat expansion events come in two forms: with explicit coordinates and allele sequences (CHROM/POS/REF/ALT),
+        # or without them. In the first case we can compute the explicit variant length as len(ALT) - len(REF). In the
+        # second case, which is more rare but still important, we have to resort to parsing HGVS-like variant names.
+        microsatellite_category = clinvar_record.measure.microsatellite_category
+        stats[microsatellite_category] += 1
+        # Skip the record if it's a deletion or a short insertion
+        if not clinvar_record.measure.is_repeat_expansion_variant:
+            continue
+
+        # Extract gene symbol(s). Here and below, dashes are sometimes assigned to be compatible with the variant
+        # summary format which was used previously.
+        gene_symbols = clinvar_record.measure.preferred_gene_symbols
+        if not gene_symbols:
+            gene_symbols = ['-']
+
+        # Extract HGNC ID
+        hgnc_ids = clinvar_record.measure.hgnc_ids
+        hgnc_id = hgnc_ids[0] if len(hgnc_ids) == 1 and len(gene_symbols) == 1 else '-'
+
+        # Append data strings
+        for gene_symbol in gene_symbols:
+            variant_data.append([
+                clinvar_record.measure.name,
+                clinvar_record.accession,
+                gene_symbol,
+                hgnc_id
+            ])
+    total_repeat_expansion_variants = clinvar_xml_utils.ClinVarRecordMeasure.MS_REPEAT_EXPANSION + \
+                                      clinvar_xml_utils.ClinVarRecordMeasure.MS_NO_COMPLETE_COORDS
+    logger.info(f'Done. A total of {i} records, {total_repeat_expansion_variants} repeat expansion variants')
```

Comment:
Just to check my understanding, technically at this point this is like "repeat expansion variant candidates", because for the `MS_NO_COMPLETE_COORDS` we don't know for sure whether they're expansions until we go through `determine_repeat_type`... is that right?

Reply:
Yes, you are very much correct. I've changed it to “repeat expansion variant candidates” to avoid possible confusion: https://github.com/EBIvariation/eva-opentargets/pull/194/commits/6b6011328a2681f606277347dfc54cd30fed2c69